### PR TITLE
[Feature] [REST-API]add rest api for runningjob detail

### PR DESCRIPTION
**Pull request by liugddx**
## Check list

* [ ] Code changed are covered with tests, or it does not need tests for reason:
* [ ] If any new Jar binary package adding in your PR, please add License Notice according
  [New License Guide](https://github.com/apache/incubator-seatunnel/blob/dev/docs/en/contribution/new-license.md)
* [ ] If necessary, please update the documentation to describe the new feature. https://github.com/apache/incubator-seatunnel/tree/dev/docs
* [ ] If you are contributing the connector code, please check that the following files are updated:
  1. Update change log that in connector document. For more details you can refer to [connector-v2](https://github.com/apache/incubator-seatunnel/tree/dev/docs/en/connector-v2)
  2. Update [plugin-mapping.properties](https://github.com/apache/incubator-seatunnel/blob/dev/plugin-mapping.properties) and add new connector information in it
  3. Update the pom file of [seatunnel-dist](https://github.com/apache/incubator-seatunnel/blob/dev/seatunnel-dist/pom.xml)
* [ ] Update the [`release-note`](https://github.com/apache/incubator-seatunnel/blob/dev/release-note.md).

### CODE REVIEW

**Comment by liugddx** [comment]
```
GET http://localhost:5801/hazelcast/rest/maps/running-jobs
```

return info
```
[
  "{\"vertices\":[{\"id\":1,\"name\":\"Console(id=1)\",\"parallelism\":2},{\"id\":2,\"name\":\"FakeSource(id=2)\",\"parallelism\":1},{\"id\":3,\"name\":\"FakeSource(id=3)\",\"parallelism\":1}],\"edges\":[{\"inputVertex\":\"FakeSource\",\"targetVertex\":\"Console\"},{\"inputVertex\":\"FakeSource\",\"targetVertex\":\"Console\"}],\"jobId\":675252797213507585}"
]
```

**Comment by liugddx** [comment]
#4044

**Comment by liugddx** [comment]
request

```
http://localhost:5801/hazelcast/rest/maps/running-jobs
```

response

```
[
  {
    "jobId": 675720885998452737,
    "jobName": "fake_to_console_1",
    "jobStatus": "RUNNING",
    "envOptions": {
      "checkpoint.interval": "5000"
    },
    "createTime": "2023-02-08 23:13:33",
    "jobDag": {
      "vertices": [
        {
          "id": 1,
          "name": "Console(id=1)",
          "parallelism": 2
        },
        {
          "id": 2,
          "name": "FakeSource(id=2)",
          "parallelism": 1
        },
        {
          "id": 3,
          "name": "FakeSource(id=3)",
          "parallelism": 1
        }
      ],
      "edges": [
        {
          "inputVertex": "FakeSource",
          "targetVertex": "Console"
        },
        {
          "inputVertex": "FakeSource",
          "targetVertex": "Console"
        }
      ]
    },
    "pluginJarsUrls": [
      
    ],
    "isStartWithSavePoint": false,
    "metrics": {
      "sourceReceivedCount": "1000005",
      "sinkWriteCount": "1000005"
    }
  }
]
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 1]
```diff
@@ -0,0 +1,137 @@
...
+        IMap<Long, JobInfo> values = this.textCommandService.getNode().getNodeEngine().getHazelcastInstance().getMap(Constant.IMAP_RUNNING_JOB_INFO);
+        Map<String, Object> extensionServices = this.textCommandService.getNode().getNodeExtension().createExtensionServices();
+        SeaTunnelServer seaTunnelServer = (SeaTunnelServer) extensionServices.get(Constant.SEATUNNEL_SERVICE_NAME);
+        JsonArray jobs = values.entrySet().stream()
+            .map(jobInfoEntry -> {
+                JsonObject jobInfo = new JsonObject();
+                JobImmutableInformation jobImmutableInformation = this.textCommandService.getNode().getNodeEngine().getSerializationService()
+                    .toObject(this.textCommandService.getNode().getNodeEngine().getSerializationService()
+                        .toObject(jobInfoEntry.getValue().getJobImmutableInformation()));
+                LogicalDag logicalDag = this.textCommandService.getNode().getNodeEngine().getSerializationService()
```
Comment (EricJoy2048) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java]:
```
        classLoader = new SeaTunnelChildFirstClassLoader(jobImmutableInformation.getPluginJarsUrls());
        logicalDag = CustomClassLoadedObject.deserializeWithCustomClassLoader(nodeEngine.getSerializationService(),
            classLoader, jobImmutableInformation.getLogicalDag());
```

**Review by EricJoy2048 — DISMISSED** [review]

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 2]
```diff
@@ -0,0 +1,209 @@
...
+    public RestHttpGetCommandProcessor(TextCommandService textCommandService) {
+        this(textCommandService, new HttpGetCommandProcessor(textCommandService));
+    }
+
+    public RestHttpGetCommandProcessor(
+            TextCommandService textCommandService,
+            HttpGetCommandProcessor httpGetCommandProcessor) {
+        super(
+                textCommandService,
+                textCommandService.getNode().getLogger(Log4j2HttpGetCommandProcessor.class));
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java:63]:
```suggestion
                textCommandService.getNode().getLogger(RestHttpGetCommandProcessor.class));
```

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/NodeExtension.java [hunk 3]
```diff
@@ -77,6 +78,7 @@ public TextCommandService createTextCommandService() {
             {
                 register(HTTP_GET, new Log4j2HttpGetCommandProcessor(this));
                 register(HTTP_POST, new Log4j2HttpPostCommandProcessor(this));
+                register(HTTP_GET, new RestHttpGetCommandProcessor(this));
```
Comment (hailin0) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/NodeExtension.java:81]: Please test `Log4j2HttpGetCommandProcessor` is normal

Comment (hailin0) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/NodeExtension.java:81]: @liugddx

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java [hunk 4]
```diff
@@ -0,0 +1,209 @@
...
+
+    private Map<String, Long> getJobMetrics(String jobMetrics) {
+        Map<String, Long> metricsMap = new HashMap<>();
+        long sourceReadCount = 0L;
+        long sinkWriteCount = 0L;
+        try {
+            JsonNode jobMetricsStr = new ObjectMapper().readTree(jobMetrics);
+            for (int i = 0; i < jobMetricsStr.get(SOURCE_RECEIVED_COUNT).size(); i++) {
+                JsonNode sourceReader = jobMetricsStr.get(SOURCE_RECEIVED_COUNT).get(i);
+                JsonNode sinkWriter = jobMetricsStr.get(SINK_WRITE_COUNT).get(i);
```
Comment (zhuangchong) [inline seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/rest/RestHttpGetCommandProcessor.java]: LGTM.

`jobMetricsStr.get(SOURCE_RECEIVED_COUNT)` and `jobMetricsStr.get(SINK_WRITE_COUNT)` can be placed outside the for loop.

File: seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/NodeExtension.java (hunk 3 above)
Comment (liugddx) [reply seatunnel-engine/seatunnel-engine-server/src/main/java/org/apache/seatunnel/engine/server/NodeExtension.java:81]: http://localhost:5801/hazelcast/rest/log-level

image

**Review by EricJoy2048 — APPROVED** [review]

**Review by zhuangchong — APPROVED** [review]
+1

**Review by Hisoka-X — APPROVED** [review]

**Pull request merged**